src/Compilers/Core/Portable/StrongName/CryptoBlobParser.cs

```diff
@@ -181,62 +190,72 @@ private static ImmutableArray<byte> CreateSnPublicKeyBlob(byte type, byte versio
         /// <remarks>
         /// Can be either a PUBLICKEYBLOB or PRIVATEKEYBLOB. The BLOB must be unencrypted.
         /// </remarks>
-        public unsafe static bool TryGetPublicKey(ImmutableArray<byte> blob, out ImmutableArray<byte> publicKey)
+        public static bool TryParseKey(ImmutableArray<byte> blob, out ImmutableArray<byte> snKey, out RSAParameters? privateKey)
         {
-            // Is this already a strong name PublicKeyBlob?
+            privateKey = null;
+            snKey = default(ImmutableArray<byte>);
+
+            var asArray = blob.ToArray();
```

Comment:
For some reason I thought `DangerousGetUnderlyingArray` was `unsafe`.  

Fixed.